qdrant-landing/content/documentation/tutorials/code-search.md

```diff
@@ -411,6 +411,15 @@ This is one example of how you can use different models and combine the results.
 In a real-world scenario, you might run some reranking and deduplication, as
 well as additional processing of the results.
 
+For example, our [code-search-demo](https://github.com/qdrant/code-search-demo) 
+extracts the method signature using the sentence transformer and highlights a 
+particular piece of code inside it, if the code specific model chooses it as 
+relevant as well.
+
+![](/documentation/tutorials/code-search/code-search-demo-example.png)
+
+Here it is - semantic code intelligence in action, in context!
+
```

Comment:
Hi @kacperlukawski , I'm not sure if my rewrite is accurate. I added some "alt text" to the image. Since this is in our documentation, style guides discourage the use of the exclamation point.

```suggestion
Our [Code Search Demo](https://github.com/qdrant/code-search-demo) 
extracts the method signature using the sentence transformer. If the code-
specific model finds it relevant, it then highlights a specific piece of code 
inside the signature.

![The Code search demo highlights a code block](/documentation/tutorials/code-search/code-search-demo-example.png)

Now you see semantic code intelligence, in action.

```

Reply:
Thank you, @mjang! I like your style more, but in my opinion the description may be misleading.

Let me describe the steps we perform in the code search demo to make it more straightforward.
1. The user sends a query.
2. That query is vectorized by both models simultaneously (NLP + code), giving us two different vectors.
3. Both vectors are used in parallel to find some relevant snippets.
4. Once we retrieve results for both vectors, we want to merge them. There are two scenarios: \ 
    1. Both methods returned different sets of documents, so we display some of them.
    2. There is an overlap between the search results. Thus, we merge overlapping snippets. 

NLP-based search is based on function signatures, but code search may return smaller pieces, such as loops. So, if we receive a particular function signature from the NLP model and part of its implementation from the code model, we merge the results and highlight the overlapping part. As a result, we display it like this:

![image](https://github.com/qdrant/landing_page/assets/2649301/719d3ea2-3d1f-483e-a1d9-d0a65f83e52e)

Currently, the description suggests that we load the results using the NLP model first and then evaluate every single result with the code model, to check if it also finds it relevant.

Reply:
Thanks for the description! (I'm tempted to use it to set up a new section, maybe titled `## The process`. If this revision is acceptable to you, let's merge. I might start a new PR based on that step-by-step description.)

```suggestion
Our [Code search demo](https://github.com/qdrant/code-search-demo) uses
both models. In the screenshot, we search for `flush of wal`. The result
shows relevant code, merged from both models. Note the highlighted
code in lines 621-629. It's where both models agree.

![Results from both models, with overlap](/documentation/tutorials/code-search/code-search-demo-example.png)

Now you see semantic code intelligence, in action.
```

Reply:
Approving with comments. @kacperlukawski is welcome to incorporate the suggestions of his choice before merging.